kv/kvserver: TestLossQuorumCauseLeaderWatcherToSignalUnavailable failed

kv/kvserver.TestLossQuorumCauseLeaderWatcherToSignalUnavailable [failed](https://tanzanite.cluster.engflow.com/invocations/default/87d8e72d-4e83-4a32-a5da-7f2f480581b0?testReportRun=2&testReportShard=15&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on release-25.2.0-rc @ [545da2936c70fdecb9bbbd652aa4bd90b8d26dad](https://github.com/cockroachdb/cockroach/commits/545da2936c70fdecb9bbbd652aa4bd90b8d26dad):


```
=== RUN   TestLossQuorumCauseLeaderWatcherToSignalUnavailable
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLossQuorumCauseLeaderWatcherToSignalUnavailable260653659
    test_log_scope.go:76: use -show-logs to present logs inline
    client_replica_test.go:2886: condition failed to evaluate within 3m45s: from client_replica_test.go:2890: range is still unavailable
    panic.go:635: -- test log scope end --
test logs left over in: outputs.zip/logTestLossQuorumCauseLeaderWatcherToSignalUnavailable260653659
--- FAIL: TestLossQuorumCauseLeaderWatcherToSignalUnavailable (246.17s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=15</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLossQuorumCauseLeaderWatcherToSignalUnavailable.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-50424

--- a/pkg/kv/kvserver/client_replica_test.go
+++ b/pkg/kv/kvserver/client_replica_test.go
@@ -2911,12 +2911,12 @@ func TestChangeReplicasGeneration(t *testing.T) {
 	assert.EqualValues(t, repl.Desc().Generation, oldGeneration+3, "\nold: %+v\nnew: %+v", oldDesc, newDesc)
 }
 
-// TestLossQuorumCauseLeaderlessWatcherToSignalUnavailable checks that if a range
-// lost its quorum, the remaining replicas in that range will have their
+// TestLossQuorumCauseLeaderlessWatcherToSignalUnavailable checks that if a
+// range lost its quorum, the remaining replicas in that range will have their
 // leaderlessWatcher indicate that the range is unavailable. Also, it checks
 // that when the range regains quorum, the leaderlessWatcher will indicate that
 // the range is available.
-func TestLossQuorumCauseLeaderWatcherToSignalUnavailable(t *testing.T) {
+func TestLossQuorumCauseLeaderlessWatcherToSignalUnavailable(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
@@ -2947,6 +2947,12 @@ func TestLossQuorumCauseLeaderWatcherToSignalUnavailable(t *testing.T) {
 				},
 			},
 			Knobs: base.TestingKnobs{
+				Store: &kvserver.StoreTestingKnobs{
+					// Make sure that we don't depend on the consistency queue
+					// running. It could cause replicas to attempt to acquire the
+					// lease, which might unintentionally unquiesce replicas. See #146188.
+					DisableConsistencyQueue: true,
+				},
 				Server: &server.TestingKnobs{
 					StickyVFSRegistry: stickyVFSRegistry,
 				},
@@ -2967,6 +2973,16 @@ func TestLossQuorumCauseLeaderWatcherToSignalUnavailable(t *testing.T) {
 	tc.AddVotersOrFatal(t, key, tc.Targets(1)...)
 	desc, err := tc.LookupRange(key)
 	require.NoError(t, err)
+
+	// Make sure that the range is up and functional.
+	// Make sure that there is a fully functioning quorum before it introduce a
+	// temporary unavailability. This deflakes the test especially for epoch
+	// leases because the node that we haven't restarted is guaranteed to not be a
+	// learner, and therefore it can campaign and unquiesce the recently restarted
+	// node.
+	_, pErr := kv.SendWrapped(context.Background(),
+		tc.GetFirstStoreFromServer(t, 0).TestSender(), putArgs(key, []byte("init")))
+	require.NoError(t, pErr.GoError())
 
 	// Randomly stop server index 0 or 1.
 	stoppedNodeInx := rand.Intn(2)

